lib/components/RunFrameForCli/LeftHeader.tsx

```diff
@@ -0,0 +1,212 @@
+      setIsError(false)
+    }
+    if (event.event_type === "EXPORT_CREATED") {
+      setNotificationMessage(`Export created: ${event.exportFilePath}`)
+      setIsError(false)
+      setisExporting(false)
+    }
+  })
+
+  useEffect(() => {
+    if (!isSaving || requestToSaveSentAt === null) return
+    const eventsSinceRequestToSave = recentEvents.filter(
+      (event) => new Date(event.created_at).valueOf() > requestToSaveSentAt,
+    )
+
+    const saveFailedEvent = eventsSinceRequestToSave.find(
+      (event) => event.event_type === "FAILED_TO_SAVE_SNIPPET",
+    )
+    const saveSuccessEvent = eventsSinceRequestToSave.find(
+      (event) => event.event_type === "SNIPPET_SAVED",
+    )
+
+    if (saveFailedEvent) {
+      setIsSaving(false)
+      setRequestToSaveSentAt(null)
+      setNotificationMessage(
+        saveFailedEvent.message ??
+          "Failed to save snippet. See console for error.",
+      )
+      console.error(saveFailedEvent.message)
+      setIsError(true)
+      if (
+        saveFailedEvent.error_code === "SNIPPET_UNSET" &&
+        saveFailedEvent.available_snippet_names
+      ) {
+        setAvailableSnippets(saveFailedEvent.available_snippet_names)
+        setIsSelectSnippetDialogOpen(true)
+      }
+    }
+    if (saveSuccessEvent) {
+      setIsSaving(false)
+      setRequestToSaveSentAt(null)
+      setNotificationMessage("Snippet saved successfully.")
+      setIsError(false)
+    }
+  }, [recentEvents, isSaving])
+
+  const triggerSaveSnippet = async () => {
+    setIsSaving(true)
+    setRequestToSaveSentAt(Date.now())
+    setNotificationMessage(null)
+    setIsError(false)
+    await pushEvent({
+      event_type: "REQUEST_TO_SAVE_SNIPPET",
+      snippet_name: snippetName,
+    } as RequestToSaveSnippetEvent)
+  }
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <div className="rf-whitespace-nowrap rf-p-2 rf-mx-1 rf-cursor-pointer rf-relative">
+          File
+        </div>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="rf-text-sm">
+        <DropdownMenuItem onSelect={triggerSaveSnippet} disabled={isSaving}>
+          {isSaving ? "Saving..." : "Push"}
+        </DropdownMenuItem>
+        <DropdownMenuSub>
+          <DropdownMenuSubTrigger className="rf-text-sm" disabled={isExporting}>
+            {isExporting ? "Exporting..." : "Export"}
+          </DropdownMenuSubTrigger>
+          <DropdownMenuPortal>
+            <DropdownMenuSubContent>
+              {availableExports.map((ext, i) => (
+                <DropdownMenuItem
+                  key={i}
+                  className="rf-text-xs"
+                  onSelect={() => {
+                    if (!isExporting) {
+                      pushEvent({
+                        event_type: "REQUEST_EXPORT",
+                        exportType: ext,
+                      })
+                      setNotificationMessage(`Export requested: ${ext}`)
```

Comment:
this is good, fwiw on the cli we will detect this and automatically trigger a download in the browser by hitting the files api, so from the user's perspective they'll just get a file download

To do that properly you would use `window.open` to the appropriate files api here, can be done in a future PR